src/main/java/com/avispl/dal/avdevices/audionetworkinterface/studiotechnologies/danteleaderclock5401a/DanteLeaderClockCommunicator.java

```diff
@@ -280,105 +307,97 @@ private void populateSyncInputDTO(Document doc, SyncInputDTO syncInputDTO) {
-						numberOfNoneNetworkProperties++;
-					}
-					networkMap.put(String.format(DanteLeaderClockConstant.TWO_STRINGS_FORMAT, propertyName, noSpaceFirstTd), tdValue);
-					index++;
+	private void networkProperties(Map<String, String> stats) throws Exception {
+		Map<String, String> networkMap = new HashMap<>();
+		Document doc = Jsoup.parse(doGet(DanteLeaderClockCommands.GET_NETWORK_COMMAND.getCommand()));
+		Elements trElements = doc.select(DanteLeaderClockConstant.TR_TAG);
+		// Remove unused tr tag
+		trElements.remove(0);
+		// List that contains network configurations is enabled
+		List<Integer> validNetworkList = new ArrayList<>();
+		// Handle the first 4 tds from the second tr to find which network properties are enabled then store to a list.
+		// If DanteSecondary is disabled => the format should look like this:
+		// ...
+		// <tr>
+		//      <td>IP Address</td>
+		//      <td>X.X.X.X</td>
+		//      <td rowspan="4">Disabled</td>
+		//      <td>X.X.X.X</td>
+		// </tr>
+		Elements tdElements = trElements.get(0).select(DanteLeaderClockConstant.TD_TAG);
+		// Populate ip addresses to stats map and add valid network properties to validNetworkList.
+		populateIpAddressProperties(networkMap, validNetworkList, tdElements);
+		int numberOfNoneNetworkProperties = 0;
+		// Loop through other tr tags to extract needed information
+		for (int i = 1; i < trElements.size(); i++) {
+			Elements tdElements2 = trElements.get(i).select(DanteLeaderClockConstant.TD_TAG);
+			// First td store the name(Subnet Mask/Gateway/MAC Address) of the property that will be extracted.
+			String firstTd = tdElements2.get(0).text();
+			int index = 1;
+			for (Integer id : validNetworkList) {
+				String tdValue = tdElements2.get(index).text();
+				String propertyName;
+				if (id == 1) {
+					propertyName = DanteLeaderClockConstant.DANTE_PRIMARY;
+				} else if (id == 2) {
+					propertyName = DanteLeaderClockConstant.DANTE_SECONDARY;
+				} else {
+					propertyName = DanteLeaderClockConstant.MANAGEMENT;
 				}
+				String noSpaceFirstTd = firstTd.replaceAll(DanteLeaderClockConstant.SPACE_REGEX, DanteLeaderClockConstant.EMPTY);
+				if (StringUtils.isNullOrEmpty(tdValue)) {
+					tdValue = DanteLeaderClockConstant.NONE;
+					numberOfNoneNetworkProperties++;
+				}
+				networkMap.put(String.format(DanteLeaderClockConstant.TWO_STRINGS_FORMAT, propertyName, noSpaceFirstTd), tdValue);
+				index++;
 			}
-			if (numberOfNoneNetworkProperties == networkMap.size()) {
-				throw new ResourceNotReachableException("Fail to populate statistics for Network group");
-			}
-			// Only put to stats here if no exception occur.
-			stats.putAll(networkMap);
-		} catch (Exception e) {
-			logger.error(e);
-			failedMonitor.put(DanteLeaderClockConstant.NETWORK, e.getMessage());
 		}
+		if (numberOfNoneNetworkProperties == networkMap.size()) {
+			throw new ResourceNotReachableException("Fail to populate statistics for Network group");
```

Comment:
and here, is there anything more to it that we could add? Like response code or something, or all we have is the fact that there's an empty network properties set? In which case, is this supposed to be an exception?

Reply:
So if we reach here then it's surely 200 OK, to make it more details let me list out the properties that are fail to be parsed or invalid format. The exception will be thrown if all fields in the network properties are None (fail to be parsed or not valid format).